src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioWorkspaceImpl.cs

```diff
@@ -644,6 +644,82 @@ protected override void ApplyAnalyzerReferenceRemoved(ProjectId projectId, Analy
             }
         }
 
+        protected override void ApplyProjectChanges(ProjectChanges projectChanges)
+        {
+            base.ApplyProjectChanges(projectChanges);
+
+            var changedMappedDocuments = projectChanges.GetChangedDocuments().Where(id => IsMappedDocument(id));
```

Comment:
I'm overriding project changes here because multiple different roslyn documents can map to the same razor document.  Getting all changes to razor documents across the project allows me to group the text changes by filepath and apply them at once with the invisible editor, instead of re-opening the invisible editor each time I encounter the same document.

Reply:
WhereAsArray.
doc this.